shared/fs/footer/upload-container.js

```diff
@@ -33,16 +34,29 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
   debugToggleShow: getDebugToggleShow(dispatch),
 })
 
-const mergeProps = ({_uploads}, {debugToggleShow}) =>
-  ({
+const mergeProps = ({_uploads, _pathItems}, {debugToggleShow}) => {
+  const folders = _pathItems.filter(path => path.type === 'folder')
+  const uploadPaths = _uploads.syncingPaths.toJS() // get array of paths
```

Comment:
I don't think this is needed. You can call `filter` directly on `_uploads.syncingPaths` which is a Set: https://facebook.github.io/immutable-js/docs/#/Set/filter